distributed/tests/test_client.py

```diff
@@ -6791,6 +6791,18 @@ def setup(self, worker=None):
         await c.register_worker_plugin(MyPlugin())
 
 
+@gen_cluster(client=True)
+async def test_scheduler_plugins(c, s, a, b):
+    class Plugin(SchedulerPlugin):
+        name = "plugin"
+
+    assert "plugin" not in s.plugins
+    await c.register_scheduler_plugin(Plugin())
+    assert "plugin" in s.plugins
+    await c.unregister_scheduler_plugin("plugin")
+    assert "plugin" not in s.plugins
+
```

Comment:
I'd suggest testing that errors in `unregister_scheduler_plugin` get propagated to the client.

```suggestion
    with pytest.raises(ValueError, match="Could not find plugin"):
        await c.unregister_scheduler_plugin(name="plugin")
```

Alternatively, this could be its own test (see `test_worker_plugin.py::test_remove_with_client_raises`)